it/src/test/java/org/eclipse/jkube/integrationtests/wildfly/jar/WildflyJar.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.integrationtests.wildfly.jar;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import org.eclipse.jkube.integrationtests.JKubeCase;
+import org.eclipse.jkube.integrationtests.maven.BaseMavenCase;
+
+import static org.eclipse.jkube.integrationtests.assertions.PodAssertion.assertPod;
+import static org.eclipse.jkube.integrationtests.assertions.PodAssertion.awaitPod;
+import static org.eclipse.jkube.integrationtests.assertions.ServiceAssertion.awaitService;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+
+abstract class WildflyJar extends BaseMavenCase implements JKubeCase {
+
+  static final String PROJECT_WILDFLY_JAR = "projects-to-be-tested/wildfly-jar/microprofile";
+
+  @Override
+  public String getProject() {
+    return PROJECT_WILDFLY_JAR;
+  }
+
+  @Override
+  public String getApplication() {
+    return "wildfly-jar-microprofile";
+  }
+
+  final Pod assertThatShouldApplyResources() throws Exception {
+    final Pod pod = awaitPod(this).getKubernetesResource();
+    assertPod(pod).apply(this).logContains("Deployed \"wildfly-jar-microprofile-0.0.0-SNAPSHOT.war\"", 60);
```

Comment:
Regarding Socket Timeouts, I'm not sure, but probably waiting for the "WildFly .* started" message may help.

There is some lag in between that message and `WildFly Full 20.0.0.Final (WildFly Core 12.0.1.Final) started in 12502ms - Started 250 of 322 services (129 services are lazy, passive or on-demand)`, I'm guessing this lag is longer in the CI clusters (since resources are scarce), especially in newer K8s/OpenShift versions.

Changing this to:
```suggestion
    assertPod(pod).apply(this)
      .logContains("Deployed \"wildfly-jar-microprofile-0.0.0-SNAPSHOT.war\"", 60)
      .logContains("started in", 15);
```
Fixed the issue for me ([CI run](https://github.com/manusa/jkube-integration-tests/runs/897274301)) for K8s cluster.

Besides this, maybe tuning a little the readiness probe can help, since maybe there is a delay between the time the pod gets deployed and the time it's marked ready. This may lead to a test failure in OpenShift environments since the IT performs the requests on the exposed route (for k8s we perform the request to the service exposed NodePort)

Reply:
I have added your suggestion + a check that we are running with WildFly Bootable JAR.The readiness initialDlay is 10sec, I didn't change that.
The last run we have 2 Openshift tests passing, k8s v1.12,0 passing (failure is openliberty). The remaining failure (with socket timeout awaiting for the Service it seems) is for K8s v1.18.6

Reply:
:( I did several runs in my branch and K8s always run green.
I'll try modifying OkHttpClient timeout values.